src/jltypes.c

```diff
@@ -2670,6 +2726,8 @@ int jl_subtype_le(jl_value_t *a, jl_value_t *b, int ta, int invariant)
                 return 0;
             }
             tta = tta->super; super = 1;
+            if (tta == NULL)
+                jl_error("cannot compute subtype relation for type under construction");
```

Comment:
I feel like this doesn't really belong as part of subtyping. I think there should be a more explicit check, traversing the supertype and looking for invalid uses.